CMakeLists.txt

```diff
@@ -26,13 +27,23 @@ if(RABIT_BUILD_TESTS)
   list(APPEND rabit_libs rabit_mock) # add to list to apply build settings, then remove
 endif()
 
-foreach(lib ${rabit_libs})
-  #include "./internal/utils.h"
-  target_include_directories(${lib} PUBLIC
-    "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>"
-    "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/dmlc-core/include>"
-    )
-endforeach()
+if(RABIT_BUILD_DMLC)
+  foreach(lib ${rabit_libs})
+    #include "./internal/utils.h"
+    target_include_directories(${lib} PUBLIC
+      "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>"
+      "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/dmlc-core/include>"
+      )
+  endforeach()
+else()
+  foreach(lib ${rabit_libs})
+    #include "./internal/utils.h"
+    target_include_directories(${lib} PUBLIC
+      "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>"
+      "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/../dmlc_core/include>"
```

Comment:
may leave comments here about we assume rabit and dmlc-core are both put in the root path of the project which depends on it?

Reply:
or we have some way to detect it

Reply:
nvm, I think they only need to be in the same level of dir which is very likely to be the case